src/main/kotlin/com/the_chance/endpoints/Products.kt

```diff
@@ -9,23 +9,67 @@ import io.ktor.server.request.*
 import io.ktor.server.response.*
 import io.ktor.server.routing.*
 
-fun Route.productsRoutes(productService: ProductService){
+fun Route.productsRoutes(productService: ProductService) {
 
-    get("/products"){
-        val products = productService.getAllProducts()
-        call.respond(ServerResponse.success(products))
-    }
+    route("/products") {
+
+        get {
+            val products = productService.getAllProducts()
+            call.respond(ServerResponse.success(products))
+        }
 
-    post("/products") {
-        val params = call.receiveParameters()
-        val productName = params["name"]?.trim().orEmpty()
-        val productPrice = params["price"]?.trim()?.toDoubleOrNull().orZero()
+        post {
+            val params = call.receiveParameters()
+            val productName = params["name"]?.trim().orEmpty()
+            val productPrice = params["price"]?.trim()?.toDoubleOrNull().orZero()
+            val productQuantity = params["quantity"]?.trim().orEmpty()
+
+            if (productName.length < 4) {
+                call.respond(ServerResponse.error("Product name length should be 4 characters or more"))
+            } else if (productPrice < 0.0) {
+                call.respond(ServerResponse.error("Product Price should be positive number."))
+            } else {
+                val newAddedProduct = productService.create(productName, productPrice, productQuantity)
+                call.respond(HttpStatusCode.Created, ServerResponse.success(newAddedProduct))
```

Comment:
Isn't good to show message that product is added successfully ?!
I won't be happy if i made a request and get a success message = null